source/tests/common/dpmodel/test_fitting_middle_output.py

```
@@ -0,0 +1,186 @@
+        nf, nloc, _ = descriptor.shape
+        fparam = np.zeros([nf, 2], dtype=np.float64)
+        aparam = np.zeros([nf, nloc, 3], dtype=np.float64)
+        ft.set_return_middle_output(True)
+        ret = ft.call(descriptor, atype, fparam=fparam, aparam=aparam)
+        self.assertIn("middle_output", ret)
+        expected_shape = (nf, nloc, ft.neuron[-1])
+        self.assertEqual(ret["middle_output"].shape, expected_shape)
+
+    def test_middle_output_registered_in_output_def(self) -> None:
+        """middle_output should appear in output_def when enabled."""
+        ft, descriptor, atype = self._build_fitting(mixed_types=True)
```

Ruff (0.15.9)

[warning] Unpacked variable `descriptor` is never used

Prefix it with an underscore or any other dummy variable pattern

(RUF059)